### PR TITLE
Gateway 3384 - AdminDistribution Standard Auditing

### PR DESCRIPTION
**Pull request by jsmith2**

### CODE REVIEW

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/transform/audit/AdminDistTransforms.java [hunk 1]
```diff
@@ -238,16 +133,67 @@ protected boolean areRequiredUserTypeFieldsNull(AssertionType oAssertion) {
...
+            } else {
+                communityId = getHCIDFromAssertion(assertion);
+            }
+        }
 
-        return bReturnVal;
+        return communityId;
     }
 
+    private String getHCIDFromAssertion(AssertionType assertion){
```
Comment (msweaver) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/transform/audit/AdminDistTransforms.java]: There is a util class that already does this. See Alamelu's pull request for details.

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/transform/audit/AdminDistTransforms.java [hunk 2]
```diff
@@ -238,16 +133,67 @@ protected boolean areRequiredUserTypeFieldsNull(AssertionType oAssertion) {
...
+    	
+    	String communityId = null;
+        boolean useTarget = false;
+
+        if (NhincConstants.AUDIT_LOG_ADAPTER_INTERFACE.equalsIgnoreCase(_interface)
+                || NhincConstants.AUDIT_LOG_ENTITY_INTERFACE.equalsIgnoreCase(_interface)) {
+            communityId = getLocalHCID();
+        } else if (NhincConstants.AUDIT_LOG_NHIN_INTERFACE.equalsIgnoreCase(_interface)) {
+            if (NhincConstants.AUDIT_LOG_OUTBOUND_DIRECTION.equalsIgnoreCase(direction)) {
+                useTarget = true;
```
Comment (akong) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/transform/audit/AdminDistTransforms.java]: you don't really need to have a useTarget boolean.  just set the communityId here using getHCIDFromTarget() and then on the else, set the communityId using getHCIDFromAssertion()

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/transform/audit/AdminDistTransforms.java [hunk 3]
```diff
@@ -238,16 +133,67 @@ protected boolean areRequiredUserTypeFieldsNull(AssertionType oAssertion) {
...
+    	String homeCommunity = null;
+    	try{
+    		homeCommunity = assertion.getHomeCommunity().getHomeCommunityId();
+    	}catch (NullPointerException ex){
+    		LOG.warn("Could not obtain HCID from HomeCommunity in assertion.", ex);
+    	}
+    	return homeCommunity;
+    }
+    
+    private String getHCIDFromTarget(NhinTargetSystemType target){
```
Comment (akong) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/transform/audit/AdminDistTransforms.java]: Use HomeCommunityMap.getCommunityIdFromTargetSystem() instead

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/transform/audit/AdminDistTransforms.java [hunk 4]
```diff
@@ -238,16 +133,67 @@ protected boolean areRequiredUserTypeFieldsNull(AssertionType oAssertion) {
...
+    	String homeCommunity = null;
+    	try{
+    		homeCommunity = target.getHomeCommunity().getHomeCommunityId();
+    	}catch(NullPointerException ex){
+    		LOG.warn("Could not obtain HCID from Target.", ex);
+    	}
+    	return homeCommunity;
+    }
+    
+    protected String getLocalHCID(){
```
Comment (akong) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/transform/audit/AdminDistTransforms.java]: Use HomeCommunityMap.getLocalHomeCommunityId()

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/transform/audit/AdminDistTransforms.java (hunk 3 above)
Comment (jsmith2) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/transform/audit/AdminDistTransforms.java]: I'm updating now for all 3 private methods that can be replaced by HomeCommunityMap.  just need to update unit test.

File: Product/Production/Services/AdminDistributionCore/src/main/java/gov/hhs/fha/nhinc/admindistribution/outbound/StandardOutboundAdminDistribution.java [hunk 5]
```diff
@@ -111,7 +111,6 @@ public void sendAlertMessage(RespondingGatewaySendAlertMessageType message, Asse
 
                 if (bIsPolicyOk) {
                     NhinTargetSystemType targetSystem = buildTargetSystem(urlInfo);
-                    auditMessage(message, assertion, NhincConstants.AUDIT_LOG_OUTBOUND_DIRECTION);
```
Comment (akong) [inline Product/Production/Services/AdminDistributionCore/src/main/java/gov/hhs/fha/nhinc/admindistribution/outbound/StandardOutboundAdminDistribution.java:114]: Is the Nhin Outbound (Requesting side) audit being done somewhere else?

Comment (jsmith2) [inline Product/Production/Services/AdminDistributionCore/src/main/java/gov/hhs/fha/nhinc/admindistribution/outbound/StandardOutboundAdminDistribution.java:114]: Yes.  This audit was an unnecessary Entity Outbound audit, was the same as EntityInbound but with different timestamp.

**Pull request merged**